service/service.go

```diff
@@ -122,11 +124,13 @@ func New(config Config) (*Service, error) {
 	var legacyController *legacy.Controller
 	{
 		c := legacy.ControllerConfig{
-			BaseDomain:       config.Viper.GetString(config.Flag.Service.Prometheus.BaseDomain),
-			K8sClient:        k8sClient,
-			Logger:           config.Logger,
-			PrometheusClient: prometheusClient,
-			Provider:         config.Viper.GetString(config.Flag.Service.Provider.Kind),
+			BaseDomain:                config.Viper.GetString(config.Flag.Service.Prometheus.BaseDomain),
+			SupportsPersistentStorage: config.Viper.GetBool(config.Flag.Service.Prometheus.Storage.IsPersistent),
```

Comment:
See note above, but my preference would be to keep this consistent everywhere and reflect what it's actually going to do, so I'd vote for renaming this to `CreatePVC`.